pkg-svcs/backend/lib/plugins/otlp-metrics/index.js

```diff
@@ -0,0 +1,51 @@
+const fp = require('fastify-plugin')
+const { enableMonitoring } = require('./setup')
+
+async function plugin(server, options) {
+  const hostname = require('os').hostname()
+  const monitoring = enableMonitoring(options.otlp)
+
+  server.decorate('monitoring', monitoring)
+
+  server.addHook('onRequest', (req, _res, done) => {
```

Comment:
nit: prefer async/await VS callback based

Reply:
Is there a reason for this preference?
I always used the callback when the function body doesn't use promises 
just to reduce the promise overhead (that is often not relevant 😄 )

Reply:
yes, it's less prone to errors because you don't have to remember to call done at the end